compiler/damlc/daml-doc/src/DA/Daml/Doc/HaddockParse.hs

```diff
@@ -388,29 +391,51 @@ getTemplateDocs DocCtx{..} typeMap templateInstanceMap =
                       [] -> [] -- catching the dummy case here, see above
                       _other -> error "getFields: found multiple constructors"
 
+-- | As per issue #2239, a template instance is desugared into a
+-- newtype with a docs marker. For example,
+--
+-- @template instance ProposalIou = Proposal Iou@
+--
+-- becomes
+--
+-- @newtype ProposalIou = MkProposalIou with unProposalIou : Proposal Iou -- ^ TEMPLATE_INSTANCE@
+--
+-- So the goal of this function is to extract the template instance doc
+-- from the newtype doc if it exists.
+getTemplateInstanceDoc :: ADTDoc -> Maybe TemplateInstanceDoc
+getTemplateInstanceDoc adt
+    | ADTDoc{..} <- adt
+    , [RecordC{..}] <- ad_constrs
+    , [FieldDoc{..}] <- ac_fields
+    , Just (DocText "TEMPLATE_INSTANCE") <- fd_descr
```

Comment:
Might be a good idea to add a comment here that this marker does not exist so far.

Reply:
Good idea :-) thanks